packages/x-data-grid-pro/src/tests/editComponents.DataGridPro.test.tsx

```diff
@@ -91,34 +91,36 @@ describe('<DataGridPro /> - Edit components', () => {
       const input = within(cell).getByRole<HTMLInputElement>('textbox');
       expect(input.value).to.equal('Nike');
 
-      fireEvent.change(input, { target: { value: 'Puma' } });
+      await user.type(input, '[Backspace>4]Puma');
       expect(input.value).to.equal('PUMA');
     });
 
     describe('with fake timers', () => {
-      clock.withFakeTimers();
-
       it('should display a indicator while processing the props', async () => {
         defaultData.columns[0].preProcessEditCellProps = ({ props }) =>
           new Promise((resolve) => {
             setTimeout(() => resolve(props), 500);
           });
-        render(<TestCase />);
+        const { user } = render(<TestCase />);
 
         const cell = getCell(0, 0);
-        fireEvent.doubleClick(cell);
+        await user.dblClick(cell);
 
         const input = within(cell).getByRole<HTMLInputElement>('textbox');
         expect(input.value).to.equal('Nike');
 
         expect(screen.queryByTestId('LoadIcon')).to.equal(null);
-        fireEvent.change(input, { target: { value: 'Puma' } });
+        await user.type(input, '[Backspace>4]Puma');
 
-        clock.tick(200);
+        await act(async () => {
+          await sleep(200);
+        });
```

Comment:
Does `sleep` always need to be wrapped like this? If so, it might make sense to do it inside `sleep` itself.

Reply:
Nice suggestion, will check feasibility in another PR